network/shuffle/src/main/java/org/apache/spark/network/shuffle/ExternalShuffleBlockResolver.java

```diff
@@ -81,7 +142,18 @@ public void registerExecutor(
       ExecutorShuffleInfo executorInfo) {
     AppExecId fullId = new AppExecId(appId, execId);
     logger.info("Registered executor {} with {}", fullId, executorInfo);
-    executors.put(fullId, executorInfo);
+    synchronized (executors) {
```

Comment:
I don't think we need to synchronize here anymore. I think this was here from when you were doing files.  leveldb write operations are atomic.

although we should do executors.put after doing db.put so we know its saved.